ci/check_configuration_keys.py

```diff
@@ -28,30 +76,38 @@ def main(values_path: str, readme_path: str) -> None:
     readme_distinct = compare_list_of_keys(readme.keys(), values_keys)
     diff_defaults = compare_values(readme, values_keys, values)
 
-    print('*' * 20)
-    print(f'Keys in values not covered by readme ({len(values_distinct)}):')
-    print('*' * 20)
-    for key in values_distinct:
-        print(key)
-
-    print('*' * 20)
-    print(f'Keys in readme not existing in values ({len(readme_distinct)}):')
-    print('*' * 20)
-    for key in readme_distinct:
-        print(key)
-
-    max_key_length = max(len(key) for key in diff_defaults.keys())
-    print('*' * 20)
-    print(f'Default values comparison ({len(diff_defaults.keys())}):')
-    print('*' * 20)
-    print(f'| {"Key":{max_key_length}} | {"Default for readme":100} | {"Default for values":100} |')
-    print(f'|{"-"*(max_key_length+2)}|{"-"*102}|{"-"*102}|')
-
-    for key in sorted(diff_defaults.keys()):
-        readme_value, values_value = diff_defaults[key]
-
-        # Show only first 100 characters of every default
-        print(f'| {key:{max_key_length}} | {readme_value[:100]:100} | {values_value[:100]:100} |')
+    if values_distinct:
+        print('*' * 20)
+        print(f'Keys in values not covered by readme ({len(values_distinct)}):')
+        print('*' * 20)
+        for key in values_distinct:
+            print(key)
+
+    if readme_distinct:
+        print('*' * 20)
+        print(f'Keys in readme not existing in values ({len(readme_distinct)}):')
+        print('*' * 20)
+        for key in readme_distinct:
+            print(key)
+
+    if diff_defaults:
+        max_key_length = max(len(key) for key in diff_defaults.keys())
+
+        print('*' * 20)
+        print(f'Default values comparison ({len(diff_defaults.keys())}):')
+        print('*' * 20)
+        if not full_diff:
+            print(f'| {"Key":{max_key_length}} | {"Default for readme":100} | {"Default for values":100} |')
+            print(f'|{"-"*(max_key_length+2)}|{"-"*102}|{"-"*102}|')
+
+        for key in sorted(diff_defaults.keys()):
+            readme_value, values_value = diff_defaults[key]
+
+            # Show only first 100 characters of every default
+            if full_diff:
```

Comment:
should we check for this outside the loop?

Reply:
`readme_value` and `values_value` is assigned in the loop. Alternatively I could create two independent loops on `for key in sorted(diff_defaults.keys()):` level. Not sure which solution is better

Reply:
Changed it and it seems to be better in terms of readability :see_no_evil: